src/keywrap/keyprovider.rs

```diff
@@ -303,7 +303,7 @@ impl KeyProviderKeyWrapper {
                 })
             }
         } else {
-            return Err(anyhow!("keyprovider: runner for binary provider is NULL"));
+            bail!("keyprovider: runner for binary provider is NULL");
```

Comment:
You may also use `anyhow!("keyprovider: runner for binary provider is NULL")` without `:`?

Reply:
Sorry I did not get your point? `bail!` here is short for `Err(anyhow!(...))`